Add marketplace recipe (#1592)
saleor/saleor-docs: docs/recipes/marketplace.mdx

    C --Attribute Reference--> F[**Vendor 2**<br/>_Saleor Model_]
    D --Attribute Reference--> G[**Vendor 3**<br/>_Saleor Model_]
    
    subgraph "Product Information"
    A
    end
    
    subgraph "Vendor Offers"
    B
    C
    D
    end
    
    subgraph "Vendors"
    E
    F
    G
    end
```

### Boutique Marketplace
A boutique marketplace, exemplified by platforms like Etsy, is simpler in structure.

In this approach, each product is unique and directly linked to its vendor without the need for a top-level product model:
- **Product**: A Saleor Product entity that contains both product information and vendor-specific details
- **Vendor**: A Saleor Model entity that contains vendor information (profile, bio, contact details)

```mermaid
graph LR
    A[**Product 1**<br/>_Saleor Product_] --Attribute Reference--> B[**Vendor 1**<br/>_Saleor Model_]
    C[**Product 2**<br/>_Saleor Product_] --Attribute Reference--> D[**Vendor 2**<br/>_Saleor Model_]
    E[**Product 3**<br/>_Saleor Product_] --Attribute Reference--> F[**Vendor 3**<br/>_Saleor Model_]
    
    subgraph "Products"
    A
    C
    E
    end
    
    subgraph "Vendors"
    B
    D
    F
    end
```

> ## Vendor Authorization
> 
> Since Saleor doesn't have a built-in vendor entity or vendor-based permissions, you need to implement a custom authorization system. The typical approach is to:
> 
> 1. Create vendors as regular staff users in Saleor with no permissions
> 2. Build a separate [vendor portal](#vendor-portal) that implements its own permission system
> 
> This approach means that while vendors exist in Saleor as staff users, you will need to handle vendor-specific access control outside of Saleor.
> 
> <RecipeResourceGrid>
>     <CompactCard
>         name="Authentication"
>         description="Learn about authentication in Saleor."
>         link="/api-usage/authentication"
>     />
> </RecipeResourceGrid>

Reviewer: ```suggestion
## Vendor Permission Management

Saleor's permission system doesn't have a built-in vendor entity or vendor-based permissions, which means it has to be implemented separately. 

Our recommendation is to integrate OIDC provider with both Saleor and [vendor portals](#vendor-portal). You can integrate with services like Auth0 or KeyCloak. Choosing this approach has multiple advantages:

- **Single source of truth for permissions:** Saleor and portal users permissions will be managed with single service 
- **Single Sign-on (SSO):** using the same accounts to access Saleor Dashboard and portals

```mermaid
sequenceDiagram
  actor vendor as Vendor
  participant vendor_portal as Vendor Portal
  participant oidc as OIDC
  participant saleor as Saleor

  vendor ->>+ vendor_portal: Request to change product
  vendor_portal ->> oidc: Checking vendor permissions
  oidc ->> vendor_portal: 
  vendor_portal ->> saleor: Product mutation using Vendor Portal token
  saleor ->> vendor_portal: 
  vendor_portal ->>- vendor: Changed product 

```

<RecipeResourceGrid>
    <CompactCard
        name="Authentication"
        description="Learn about authentication in Saleor."
        link="/api-usage/authentication"
    />
    <CompactCard
        name="OIDC Plugin"
        description="Integrating Saleor with external identity providers."
        link="/developer/app-store/plugins/oidc"
    />
</RecipeResourceGrid>

```

WDYT? In such platforms SSO is often also implemented to provide social accounts login to the platform. Instead of fragmented permissions it can be cleanly managed in a single place.

1. Integrate with the Saleor API
2. Implement custom authentication for marketplace operators
3. Provide vendor management features:
   - Vendor onboarding workflow
   - Policy management
   - Commission fee configuration
   - Marketplace health monitoring
   - Vendor performance tracking

The operator portal is essential for marketplace governance and cannot be implemented using the standard Saleor dashboard. It requires custom development to handle the complex requirements of marketplace administration and oversight.

Operator Portal can be a Dashboard Extension or a separate application.

<RecipeResourceGrid>
    <CompactCard
        name="Extending Dashboard"
        description="Learn how to extend the Saleor dashboard."
        link="/recipes/extending-dashboard"
    />    
</RecipeResourceGrid>

## Architecture Overview

Here is a high-level overview of the essential services for a Saleor-based marketplace:

> ```mermaid
> architecture-beta
>   service storefront(internet)[Storefront]
>   service saleor(server)[Saleor]
>   service vendor_portal(cloud)[Vendor Portal]
>   service operator_portal(cloud)[Operator Portal]
>   service payment_service(cloud)[Payment Service]
> 
>   storefront:B -- T:saleor
>   vendor_portal:R -- L:saleor
>   payment_service:L -- R:saleor
>   saleor:B -- T:operator_portal
> ```

Reviewer: ```suggestion
```mermaid
architecture-beta
  service storefront(internet)[Storefront]
  service vendor_portal(cloud)[Vendor Portal] 
  service operator_portal(cloud)[Operator Portal] 
  service payment_service(cloud)[Payment Service]
  service saleor(server)[Saleor]
  service oidc(cloud)[OpenID Provider]

  junction junctionMarketplace

  storefront:B -- T:saleor
  payment_service:L -- R:saleor
  vendor_portal:R -- L:junctionMarketplace
  operator_portal:L -- R:junctionMarketplace
  junctionMarketplace:T -- B:saleor
  oidc:R -- L:saleor
```
```

Added OIDC

> **Service Descriptions:**
> 
> - **Storefront:** The customer-facing shop interface.
> - **Vendor Portal:** For vendors to manage their profiles, publish products, configure shipping, and view their own sales data.
> - **Saleor:** The core platform, handling products, orders, and API logic.
> - **Payment Service:** Handles payment processing, refunds, integration with providers (e.g., Stripe), and payouts to vendors after sales.
> - **Operator Portal:** For marketplace operators to onboard/manage vendors and oversee the marketplace.

Reviewer: ```suggestion
**Service Descriptions:**

- **Saleor:** The core platform, handling products, orders, and API logic.
- **Storefront:** The customer-facing shop interface.
- **OIDC:** Provides centralized permission management.
- **Vendor Portal:** For vendors to manage their profiles, publish products, configure shipping, and view their own sales data.
- **Payment Service:** Handles payment processing, refunds, integration with providers (e.g., Stripe), and payouts to vendors after sales.
- **Operator Portal:** For marketplace operators to onboard/manage vendors and oversee the marketplace.
```
